### ISSUE TITLE
Enabling flash servers for legacy users

### ISSUE THREAD

**Issue by meobu** [haxball/haxball-issues#2125]
Dear @basro , this is Punisher from Big 4v4 community.

I hope this message finds you well. I am reaching out to discuss a matter of great importance for the Haxball community, particularly those who wish to relive the old-fashioned gaming experience that the Flash version provided.

As you are aware, the HTML version was introduced as a response to the discontinuation of Flash, and while it brought many improvements, there remains a sentiment among players who long for the classic Haxball experience. During the transitional phase, some enthusiasts like me and other guys were still able to utilize Flash by using older browsers, sustaining a semblance of the original gameplay.

However, it has come to our attention that the Flash servers were eventually shut down, impeding the possibility of creating rooms and generating links for gameplay. In light of this, we propose the consideration of reopening the Flash servers to cater to a niche yet passionate audience.

By reinstating the Flash servers, players who yearn for the old-school Haxball experience can use older browsers to create rooms and engage in tournaments without the interference of extrapolation. This alternative approach will not in any way detract from the ongoing success and development of the HTML version but will instead provide a distinct and cherished option for those seeking a nostalgic gaming atmosphere.

Especially within the context of larger 4v4 leagues, we believe that there is a significant audience eager to embrace this opportunity. The proposal aligns with the ethos of offering diversity in gameplay options and could potentially rejuvenate interest among players who cherish the memories associated with the Flash version.

We understand the challenges and limitations the development faces, and we sincerely appreciate the ongoing efforts to enhance the Haxball experience. This request is made in the spirit of collaboration and the shared love for the game, and we hope that you will consider the reopening of Flash servers as a gesture to the passionate Haxball community.

Thank you for your time and consideration.

Best regards.

**Comment by Darida8** [comment]
Agreed.

There is a significant decrease in number of players recently, one of them is me.

I am member of almost all big platforms in Haxball and as I observed all communities, all leagues, tournaments becoming more and more passive each day, they almost shut down.
I believe bringing Flash version back will be the key in this manner.

The reason most players quit this game is that HTML brought a handicap between those players who have a decent system & monitor , and those who don't.

It is unfair in gameplay, very hard to compete if you don't have well equipments because you play with unacceptably low FPS.

Long story short, bring Flash back and watch the resurrection of this game.

**Comment by basro** [comment]
Sorry, I will not be doing this.

Even if my servers were back online the game would still not work because it depended on a free service from adobe which has also been discontinued.

**Comment by meobu** [comment]
![image](https://github.com/haxball/haxball-issues/assets/8561445/52fe2810-dd1b-4e89-9b2f-4d618a8b22c7)

For your consideration, I am able to run the Flash version of the game through an application. However, it is encountering an error that persists even after attempting to bypass it. Furthermore, due to the servers being inactive, I will be playing alone without the possibility of multiplayer interaction.

If you can send me the non-domain locked SWF file of the Flash Haxball, I can test it at least. If it's playable, maybe you can turn on the servers, and it's a win-win situation.

This way, you won't need to develop an application because I am just using a public emulator, and that's it. The same emulator I tried Transformice's swf file and it worked flawlessly on my end. I think the same will happen for Haxball too.

I will be waiting for you. Here is my contact information:

[desyncfdp@gmail.com](mailto:desyncfdp@gmail.com)
or add from Discord: esoterik1

@basro

**Comment by meobu** [comment]
Okay, somehow I got the SWF file and decrypted the contents of it so means I got full source code and tried to play. It's flawlessly working on my end. If you can re-open the servers I will be testing it with friends to see how multiplayer is feeling. If you are not willing to reopen (which I respect), I will try to make it on my own..

Also this made me smile XD: 

```
  if(outStream != null && param1 != null)
         {
            Boot.lastError = new Error();
            throw "Shit";
         }
```

**Comment by reyeso23** [comment]
I agree with @meobu , especially that recently there have been some methods to run an SWF file easily on websites, such as "Ruffle", this won't affect the HTML version.
@meobu  can you send me the swf ?
